### PR TITLE
Fix: Returns the tokens of the position owner to the position owner and not the pool admin

### PR DESCRIPTION
**Pull request by ogarciarevett**
This pull request includes significant changes to the `Staker` contract, updates to the Hardhat configuration, and modifications to the test suite. The most important changes focus on adding an `administrator` parameter to staking pools, ensuring correct fund transfers, and enhancing the test coverage.

### Contract Changes:
* Added `administrator` parameter to the `createPool` function and updated related events and internal logic to use this parameter instead of `msg.sender`. (`web3/contracts/staking/Staker.sol`, [[1]](diffhunk://#diff-e126d5ee9d1e167d4a24fb644a8d69e9b8ba58ff99a1dd4e4ccb781faf82eb52L139-R140) [[2]](diffhunk://#diff-e126d5ee9d1e167d4a24fb644a8d69e9b8ba58ff99a1dd4e4ccb781faf82eb52L162-R163) [[3]](diffhunk://#diff-e126d5ee9d1e167d4a24fb644a8d69e9b8ba58ff99a1dd4e4ccb781faf82eb52L172-R173)
* Corrected the recipient of transfers during unstaking to use `positionOwner` instead of `msg.sender`. (`web3/contracts/staking/Staker.sol`, [web3/contracts/staking/Staker.solL460-R467](diffhunk://#diff-e126d5ee9d1e167d4a24fb644a8d69e9b8ba58ff99a1dd4e4ccb781faf82eb52L460-R467))

### Configuration Changes:
* Added new Hardhat plugins for enhanced functionality and ensured the deployer private key is required. (`web3/hardhat.config.ts`, [web3/hardhat.config.tsR3-R18](diffhunk://#diff-617c9f5b6ca0428a62dc10e440afb7cad076b43e3ead254aaea42589a88a6f7cR3-R18))
* Updated network configurations to use the deployer private key for account management. (`web3/hardhat.config.ts`, [web3/hardhat.config.tsR67-R82](diffhunk://#diff-617c9f5b6ca0428a62dc10e440afb7cad076b43e3ead254aaea42589a88a6f7cR67-R82))

### Test Suite Changes:
* Updated test cases to include the `administrator` parameter when creating staking pools. (`web3/test/Staker.test.1.ts`, [[1]](diffhunk://#diff-19454ecab72fa7ffca1bfb700ab4fd1f7d410b2aee7ec561d4e2b3b1d5fccf34L39-R40) [[2]](diffhunk://#diff-19454ecab72fa7ffca1bfb700ab4fd1f7d410b2aee7ec561d4e2b3b1d5fccf34L49-R51) [[3]](diffhunk://#diff-19454ecab72fa7ffca1bfb700ab4fd1f7d410b2aee7ec561d4e2b3b1d5fccf34L59-R62) [[4]](diffhunk://#diff-19454ecab72fa7ffca1bfb700ab4fd1f7d410b2aee7ec561d4e2b3b1d5fccf34L70-R74)
* Enhanced tests to verify correct handling of the administrator and fund transfers. (`web3/test/Staker.test.3.ts`, [[1]](diffhunk://#diff-448d049fe5e354fb4d0e36c24c2f90da982ddc6cbdccafd3d9406188f1b81775L745-R746) [[2]](diffhunk://#diff-448d049fe5e354fb4d0e36c24c2f90da982ddc6cbdccafd3d9406188f1b81775R1162-R1163) [[3]](diffhunk://#diff-448d049fe5e354fb4d0e36c24c2f90da982ddc6cbdccafd3d9406188f1b81775L1170-R1182)

These changes collectively improve the flexibility and security of the staking contract and ensure the test suite accurately reflects the new logic.

### CODE REVIEW

File: web3/hardhat.config.ts [hunk 1]
```diff
@@ -1,8 +1,9 @@
 import { HardhatUserConfig } from 'hardhat/config';
 import '@nomicfoundation/hardhat-toolbox';
+import "@nomicfoundation/hardhat-ignition-ethers";
```
Comment (zomglings) [inline web3/hardhat.config.ts:3]: What is this import doing? Are you using it @ogarciarevett ? I'm not seeing any place in the code where this is being used.

File: web3/contracts/staking/Staker.sol [hunk 2]
```diff
@@ -181,7 +182,7 @@ contract Staker is ERC721Enumerable, ReentrancyGuard {
      *
      * @param poolID The ID of the staking pool to update configuration of.
      * @param changeTransferability Specifies whether the current call is updating the transferability of the pool or not. If this is false, then the value of the `transferable` argument will be ignored.
-     * @param transferable Whether or not the pool should be transferable. This argument is only applied if `changeTransferabiliy` is `true`.
+     * @param transferable Whether or not the pool should be transferable. This argument is only applied if `changeTransferability` is `true`.
```
Comment (zomglings) [inline web3/contracts/staking/Staker.sol:185]: Thank you

**Review by zomglings — CHANGES_REQUESTED** [review]
Can you add tests for the `administrator` argument in `createPool`?

Currently, I am not seeing tests where you exercise this parameter.

There should be at least one test where you `createPool` and atomically transfer to the `0` address. This will communicate that `0` is a valid argument (there should never be any validation on the `administrator` arg).

There should be a test where you transfer administration to the `Staker` contract itself.

Also, please run `make rebuild` on this repo. You will need to install [`seer`](https://github.com/G7DAO/seer) first.

File: web3/hardhat.config.ts (hunk 1 above)
Comment (ogarciarevett) [reply web3/hardhat.config.ts:3]: its required to deploy using the latest version of hardhat, but its optional, I can remove it

**Review by zomglings — APPROVED** [review]
lgtm

**Pull request merged**